### PR TITLE
Move change_id_index() from Revset to Index

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: lib/src/default_index/revset_engine.rs [hunk 1]
```diff
@@ -133,18 +130,6 @@ where
         Box::new(self.iter_graph_impl())
     }
 
-    fn change_id_index(&self) -> Box<dyn ChangeIdIndex + 'index> {
-        // TODO: Create a persistent lookup from change id to commit ids.
```
Comment (yuja) [inline lib/src/default_index/revset_engine.rs:136]: nit: this TODO is still valid, but we have another comment about that in `CommitGraphEntry`, so it should be okay to delete now.

(no action needed)

**Review by yuja — APPROVED** [review]
Thanks.

Maybe we can now remove `'index` from the Revset trait. If you haven't started, I'll take a look.

**Comment by martinvonz** [comment]
> Maybe we can now remove `'index` from the Revset trait. If you haven't started, I'll take a look.

I haven't started, so feel free (my next little project was going to be to add a virtual root operation). Thank you.

File: lib/src/default_index/revset_engine.rs (hunk 1 above)
Comment (martinvonz) [reply lib/src/default_index/revset_engine.rs:136]: I actually meant to keep the comment. I've added it back in readonly.rs

**Pull request merged**